### PR TITLE
Add basic testing for "infer_pieces.py" module

### PR DESCRIPTION
**Pull request by David-davidlxl**
Basic testing for the "infer_pieces.py" module has been implemented (https://github.com/davidmallasen/LiveChess2FEN/issues/15).

### CODE REVIEW

**Comment by David-davidlxl** [comment]
image

File: test/test_infer_pieces.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+    :return: Length-64 list of piece probabilities.
+
+        Each element in the list is a length-13 sublist that corresponds
+        to a unique square on the chessboard.
+
+        Each sublist contains 13 piece probabilities (in the order of
+        `_IDX_TO_PIECE_FULL`) for the corresponding square.
+
+        The probabilities are hardcoded such that they are all 0s and
```
Comment (davidmallasen) [inline test/test_infer_pieces.py:32]: Could you add in the doc that the one will be exactly the piece set by the fen string?

**Review by davidmallasen — CHANGES_REQUESTED** [review]
Thanks for the PR @David-davidlxl ! Could you update the requirements file with pytest?

File: test/test_infer_pieces.py (hunk 1 above)
Comment (David-davidlxl) [reply test/test_infer_pieces.py:32]: Done! Lmk what you think

Comment (davidmallasen) [reply test/test_infer_pieces.py:32]: I would simplify it a bit more. How about:
The probabilities are hardcoded such that each sublist has exactly twelve 0s and one 1. The 1 will correspond to the piece type given by the fen string.

**Comment by davidmallasen** [comment]
When running pytest with the python versions in `requirements_pc.txt` I get a deprecation warning that you don't:
~~~
====================================================================================== warnings summary ======================================================================================
venv/lib/python3.10/site-packages/tensorflow/__init__.py:29
LiveChess2FEN/venv/lib/python3.10/site-packages/tensorflow/__init__.py:29: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
    import distutils as _distutils

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================ 1 passed, 1 warning in 1.95s ================================================================================

~~~
Do you have a different version of tensorflow installed? It should be 2.12 according to the requriements

File: test/test_infer_pieces.py (hunk 1 above)
Comment (David-davidlxl) [reply test/test_infer_pieces.py:32]: How does this sound?
image

**Comment by David-davidlxl** [comment]
> When running pytest with the python versions in `requirements_pc.txt` I get a deprecation warning that you don't:
> 
> ```
> ====================================================================================== warnings summary ======================================================================================
> venv/lib/python3.10/site-packages/tensorflow/__init__.py:29
> LiveChess2FEN/venv/lib/python3.10/site-packages/tensorflow/__init__.py:29: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
>     import distutils as _distutils
> 
> -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
> ================================================================================ 1 passed, 1 warning in 1.95s ================================================================================
> ```
> 
> Do you have a different version of tensorflow installed? It should be 2.12 according to the requriements

(I think you meant "package versions" versus "python versions btw)

I definitely have the 2.12.0 version of `tensorflow`. Here's a detailed from-installation-to-demo video I recorded with OBS Studio (sorry for the relatively low resolution--GitHub has file-size restrictions):

https://github.com/davidmallasen/LiveChess2FEN/assets/51491203/b680c436-b7b2-43e8-b953-b9e20d816a8d

**Comment by davidmallasen** [comment]
Thanks for the details @David-davidlxl , all looks correct in the video, so it may be some differences between windows and ubuntu. I saw that you install the python packages using pip directly instead of using the requirements file. To have replicable setups I would suggest the use of the requirements file always unless we want to update the versions or add new packages like now. Since everything seems fine, I'll merge the PR and I can debug that deprecation warning later.

**Review by davidmallasen — APPROVED** [review]

**Pull request merged**